src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/EventSourceTests.cs

```diff
@@ -108,5 +106,76 @@ public async Task SendingRequestProducesEvents()
             Assert.AreEqual(requestId, e.GetProperty<string>("requestId"));
             CollectionAssert.AreEqual(new byte[] { 6, 7, 8, 9, 0 }, e.GetProperty<byte[]>("content"));
         }
+
+        [Test]
+        public async Task NonSeekableResponsesAreLoggedInBlocks()
+        {
+            var mockResponse = new MockResponse(500);
+            mockResponse.ResponseContentStream = new NonSeekableMemoryStream(new byte[] { 6, 7, 8, 9, 0 });
+            var mockTransport = new MockTransport(mockResponse);
+
+            var pipeline = new HttpPipeline(mockTransport, new []{ LoggingPolicy.Shared });
+            string requestId;
+            var buffer = new byte[10];
+
+            using (var request = pipeline.CreateRequest())
+            {
+                request.SetRequestLine(HttpPipelineMethod.Get, new Uri("https://contoso.a.io"));
+                request.Content = HttpPipelineRequestContent.Create(new byte[] { 1, 2, 3, 4, 5 });
+                requestId = request.RequestId;
+
+                var response = await pipeline.SendRequestAsync(request, CancellationToken.None);
+
+
+                Assert.AreEqual(3, await response.ContentStream.ReadAsync(buffer, 5, 3));
+                Assert.AreEqual(2, await response.ContentStream.ReadAsync(buffer, 8, 2));
+                Assert.AreEqual(0, await response.ContentStream.ReadAsync(buffer, 0, 5));
+            }
+
+            EventWrittenEventArgs[] contentEvents = _listener.EventsById(11).ToArray();
```

Comment:
Is there a constant or other semantically meaningful symbol that we could use to alias the magic `11` to offer readers more context?

Reply:
I can extract it to a local constant but having `Assert.AreEqual("ResponseContentBlock",  contentEvents[0].EventName);` couple lines below usually does the trick for me.